src/coreclr/vm/jitinterface.cpp

```diff
@@ -1721,6 +1721,51 @@ bool CEEInfo::isFieldStatic(CORINFO_FIELD_HANDLE fldHnd)
     return res;
 }
 
+int CEEInfo::getArrayLength(CORINFO_OBJECT_HANDLE objHnd)
+{
+    CONTRACTL {
+        THROWS;
+        GC_TRIGGERS;
+        MODE_PREEMPTIVE;
+    } CONTRACTL_END;
+
+    int arrLen = -1;
+    JIT_TO_EE_TRANSITION();
+
+    _ASSERT(objHnd != NULL);
+
+    GCX_COOP();
+
+    Object* obj = nullptr;
+    size_t handle = (size_t)objHnd;
+    if (handle & 1)
+    {
+        obj = *(Object**)(handle - 1);
+    }
+    else
+    {
+        obj = (Object*)handle;
+        _ASSERT(GCHeapUtilities::GetGCHeap()->IsInFrozenSegment(obj));
+    }
+    _ASSERT(obj != nullptr);
```

Comment:
This should be extracted into a helper method and used to convert `CORINFO_OBJECT_HANDLE` to `OBJECTREF` everywhere (e.g. in printObjectDescription)